src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -187,48 +185,39 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
         {
             Debug.Assert(_builtApplication is not null);
 
+            // UseRouting called before WebApplication such as in a StartupFilter
+            // lets remove the property and reset it at the end so we don't mess with the routes in the filter
+            if (app.Properties.TryGetValue(EndpointRouteBuilderKey, out var priorRouteBuilder))
+            {
+                app.Properties.Remove(EndpointRouteBuilderKey);
+            }
+
             if (context.HostingEnvironment.IsDevelopment())
             {
+                // TODO: add test for this
```

Comment:
We don't have coverage for this?

Reply:
https://github.com/dotnet/aspnetcore/issues/35423